addons/manager.go

```diff
@@ -203,19 +189,13 @@ func runHubManager(ctx context.Context, options AddonAgentOptions, logger *slog.
 }
 
 func runSpokeManager(ctx context.Context, options AddonAgentOptions, logger *slog.Logger) {
-	hubConfig, err := GetClientConfig(options.HubKubeconfigFile)
+	spokeKubeConfig, err := utils.GetClientConfig(options.KubeconfigFile)
```

Comment:
What happens if this is not explicitly provided or is an invalid config?  It will error out and exit but we should  try to use default ctrl config as well

Reply:
Also will ensure to avoid failures during upgrades

Reply:
If file is not provided, we fall back to default in-cluster config. If invalid config is provided, we crash out until it is fixed, since it is an irrecoverable error.